Firestore/Protos/nanopb_cpp_generator.py

```diff
@@ -246,16 +329,308 @@ def nanopb_write(results):
   response = plugin_pb2.CodeGeneratorResponse()
 
   for result in results:
-    f = response.file.add()
-    f.name = result['headername']
-    f.content = result['headerdata']
+    base_filename = result['headername'].replace('.nanopb.h', '')
+    file_printers = pretty_printing_info[base_filename]
 
-    f = response.file.add()
-    f.name = result['sourcename']
-    f.content = result['sourcedata']
+    generate_header(response.file, result['headername'], result['headerdata'],
+                    file_printers)
+    generate_source(response.file, result['sourcename'], result['sourcedata'],
+                    file_printers)
 
   return response
 
 
+def begin_namespace(files, file_name):
+  """Opens the `firebase::firestore` namespace in file with given `file_name`.
+  """
+  f = create_insertion(files, file_name, 'includes')
+  f.content = '''\
+namespace firebase {
+namespace firestore {\n\n'''
+
+
+def end_namespace(files, file_name):
+  """Closes the `firebase::firestore` namespace in file with given `file_name`.
+  """
+  f = create_insertion(files, file_name, 'eof')
+  f.content = '''\
+}  // namespace firestore
+}  // namespace firebase\n\n'''
+
+
+def indent(level):
+  """Returns leading whitespace corresponding to the given indentation `level`.
+  """
+  indent_per_level = 4
+  return ' ' * (indent_per_level * level)
+
+
+def fixup(file_contents):
+  """Applies fixups to generated Nanopb code.
+
+  This is for changes to the code, as well as additions that cannot be made via
+  insertion points. Current fixups:
+  - rename fields named `delete` to `delete_`, because it's a keyword in C++.
+  """
+
+  delete_keyword = re.compile(r'\bdelete\b')
+  return delete_keyword.sub('delete_', file_contents)
+
+
+def create_insertion(files, file_name, insertion_point):
```

Comment:
This file-list, file name pair, along with behavior related to it seems like it could be a class.

Here's an example of what I mean:

```python
class CurrentFile(object):

  def __init__(self, files, file_name) ...

  def create_insertion(self, insertion_point):
    f = self.files.add()
    f.name = self.file_name
    f.insertion_point = insertion_point
    return f
```

I think you likely could also move `add_contents` to be a member (though I think you should name it `set_contents`, because there can only be one main, right?

Also, all of the callers of this API immediately just call `f.content =`. Consider making this method `insert`, i.e.

```python
  def insert(self, insertion_point, content):
    f = self.files.add()
    f.name = self.file_name
    f.insertion_point = insertion_point
    f.content = content
```